kv/dist_sender.go

```diff
@@ -348,8 +347,7 @@ func (ds *DistSender) getRangeDescriptors(key proto.Key, options lookupOptions)
 	} else {
 		// Look up desc from the cache, which will recursively call into
 		// ds.getRangeDescriptors if it is not cached.
-		desc, err = ds.rangeCache.LookupRangeDescriptor(metadataKey, options)
-		if err != nil {
+		if desc, err = ds.rangeCache.LookupRangeDescriptor(metadataKey, options); err != nil {
```

Comment:
This reads like a shadowing statement when it is actually setting a variable in the outer scope. I'd prefer not to embed an assignment in the if statement when we're not actually using the if statement as a scope.